File: 3875-construct-uniform-parity-array-i/3875-construct-uniform-parity-array-i.java

class Solution {
    public boolean uniformArray(int[] nums1) {
        boolean hasOdd = false;
        boolean hasEven = false;

        for (int num : nums1) {
            if (num % 2 == 1) {
                hasOdd = true;
            } else {
                hasEven = true;
            }
        }

        if (!hasOdd || !hasEven) {
            return true;
        }

        return true;
    }
}

File: 3875-construct-uniform-parity-array-i/3875-construct-uniform-parity-array-i.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums1) {
        int even = 0, odd = 0;
        int n = nums1.size();
        vector<int> nums2(n);

        int oddIndex = -1;

        for (int i = 0; i < n; i++) {
            if (nums1[i] % 2 == 0) {
                even++;
            } else {
                odd++;
                if (oddIndex == -1) {
                    oddIndex = i;
                }
            }
        }

        
        if (oddIndex != -1) {
            for (int i = 0; i < n; i++) {
                if (nums1[i] % 2 == 0) {
                    nums2[i] = nums1[i] - nums1[oddIndex]; 
                } else {
                    nums2[i] = nums1[i]; 
                }
            }
            return true;
        }

       
        return true;
    }
};